tests/utils/stl/test/tests.py

```diff
@@ -7,215 +7,199 @@
-    def writeJUnitXML(self, fil):
-        """Write the test's report xml representation to a file handle."""
-        test_name = quoteattr(self.getXMLOutputTestName())
-        class_name = quoteattr(self.getXMLOutputClassName())
-
-        testcase_template = \
-            '<testcase classname={class_name} name={test_name} ' \
-            'time="{time:.2f}"'
-        elapsed_time = self.result.elapsed if self.result.elapsed else 0.0
-        testcase_xml = \
-            testcase_template.format(class_name=class_name,
-                                     test_name=test_name,
-                                     time=elapsed_time)
-        fil.write(testcase_xml)
-
-        if self.result.code.isFailure:
-            fil.write(">\n\t<failure><![CDATA[")
-            if isinstance(self.result.output, str):
-                encoded_output = self.result.output
-            elif isinstance(self.result.output, bytes):
-                encoded_output = self.result.output.decode("utf-8", 'replace')
-            # In the unlikely case that the output contains the CDATA
-            # terminator we wrap it by creating a new CDATA block
-            fil.write(encoded_output.replace("]]>", "]]]]><![CDATA[>"))
-            fil.write("]]></failure>\n</testcase>")
-        elif self.result.code == UNSUPPORTED:
-            unsupported_features = self.getMissingRequiredFeatures()
-            if unsupported_features:
-                skip_message = \
-                    "Skipping because of: " + ", ".join(unsupported_features)
-            else:
-                skip_message = "Skipping because of configuration."
-            skip_message = quoteattr(skip_message)
-
-            fil.write(
-                ">\n\t<skipped message={} />\n</testcase>".format(
-                    skip_message))
-        elif self.result.code == SKIPPED:
-            message = quoteattr('Test is explicitly marked as skipped')
-            fil.write(">\n\t<skipped message={} />\n</testcase>".format(
-                message))
-        else:
-            fil.write("/>")
-
-
-class LibcxxTest(STLTest):
-    def getOutputBaseName(self):
-        output_base = self.path_in_suite[-1]
-
-        if output_base.endswith('.cpp'):
-            return output_base[:-4]
-        else:
-            return output_base
+            targetArch = litConfig.target_arch.casefold()
+            if (targetArch == 'x64'.casefold()):
+                self.compileFlags.append('-m64')
+            elif (targetArch == 'x86'.casefold()):
+                self.compileFlags.append('-m32')
+
+        self.cxx = os.path.normpath(cxx)
+
+        foundStd = False
+        for flag in chain(self.flags, self.compileFlags, self.linkFlags):
+            if flag[1:5] == 'std:':
+                foundStd = True
+                if flag[5:] == 'c++latest':
+                    Feature('c++2a').enableIn(self.config)
+                    self.compileFlags.append('/D_LIBCPP_CONSTEXPR_AFTER_CXX17=constexpr')

```

Comment:
We should not be defining this. If a test depends on the value of a libc++-internal macro, it's a bug in the test that needs to be fixed. (Several occurrences.)

FWIW, grepping the current llvm-project submodule I find occurrences of `_LIBCPP_CONSTEXPR_AFTER_CXX17` only in product headers.

Reply:
Hmmm, when I was running the tests I was having failures without this. Will try again.

Reply:
Aha! I thought this sounded familiar: https://github.com/llvm/llvm-project/commit/24ab5787b44435d0c7e805a8579753e48be9f5b9.